lib/search_solr_tools/helpers/harvest_status.rb

```diff
@@ -0,0 +1,66 @@
+module SearchSolrTools
+  module Helpers
+    class HarvestStatus
+
+      INGEST_OK = :ok
+      HARVEST_NO_DOCS = :harvest_none
+      HARVEST_FAILURE = :harvest_fail
+      INGEST_ERR_INVALID_DOC = :invalid
+      INGEST_ERR_SOLR_ERROR = :solr_error
+      OTHER_ERROR = :other
+      PING_SOLR = :ping_solr  # used for initialize only
+      PING_SOURCE = :ping_source  # used for initialize only
+
+      ERROR_STATUS = [HARVEST_NO_DOCS, HARVEST_FAILURE, INGEST_ERR_INVALID_DOC, INGEST_ERR_SOLR_ERROR, OTHER_ERROR]
+
+      # init_info is an optional hash that contains the various status keys and the documents to
+      # associate with them
+      def initialize(init_info={})
+        @status = { INGEST_OK => [] }
+        @ping_solr = true
+        @ping_source = true
+        ERROR_STATUS.each { |s| @status[s] = [] }
+
+        init_info.each do |key, docs|
+          @status[key] = docs if @status.include? key
+        end
+
+        @ping_solr = init_info[PING_SOLR] if init_info.include? PING_SOLR
+        @ping_source = init_info[PING_SOURCE] if init_info.include? PING_SOURCE
+      end
+
+      def record_multiple_document_status(documents, doc_status)
+        documents.each { |d| record_document_status d, doc_status }
+      end
+
+      def record_document_status(document, doc_status)
+        @status[doc_status] << document
```

Comment:
Is this saving the entire document, for every document?

Reply:
I think what you want to be doing here is just incrementing a count, not saving the document in an array.

Reply:
I guess since JSON is harvested for NSIDC, not XML, it won't be too large -- but it still seems inefficient to save the whole document if you don't actually need to use it.

Reply:
Good point.  In my head, I kept thinking of just saving the identifiers, but never made it do that.  I'll see if I can get that easily; otherwise, I'll just do the count like you say.  (Having a list of identifiers could be useful).

If nothing else, instead of "document" maybe I'll just call it "message" so that it can describe what went wrong (like it does for the errors of getting no results, where there is no "document")

Reply:
Hopefully you mean you would pass a different variable called `message` (without the full document content), not just rename the `document` variable. I think you're already passing a "message" (at least by proxy) courtesy of the `doc_status` value. I couldn't find anywhere that the stored documents are used, so a simple count is all you need.

Reply:
"Having a list of identifiers could be useful": Yes, but no use for them at this moment that I can see.

Reply:
One more note: It looks like insert_solr_doc is called with a hash of documents in some locations (auto_suggest.rb, line 49) and with a single item in others (base.rb, line 120). Could you confirm whether this is really happening, and if so, whether it's a problem?